### ISSUE TITLE
Absence of value

### ISSUE THREAD

**Issue by iw2rmb** [firebase/polymerfire#121]
### Description
Get document at any inexisting path

### Expected outcome
`null`
In firebase.js `snapshot.val()` returns `null` if value is empty.

### Actual outcome
`{}`

### Browsers Affected

- [X] Chrome

**Comment by tjmonsi** [comment]
Based from what I have experienced working on this, it forces all values to be {} and not null. There's this line in https://github.com/firebase/polymerfire/blob/master/firebase-document.html at 163 where it uses `this.zeroValue`. If you look at line 66. `get zeroValue()` returns `{}`. You might want to set zeroValue to null if you want (but I haven't tried that though). The problem with doing a null for zeroValue is it might have undesired consequences, like accidentally deleting the value of the path (because saving at a path with a null value deletes it). 

That's why what I did for now (which is a bit hackish) is to use a checker to check if the value returned is an empty object. signifying that it is either null or {}, which is for me a false value if I want to check for its existence

**Comment by iw2rmb** [comment]
@tjmonsi thank you! You're right. Seems very risky. But I'll try to experiment with it.
I'm having some mess with code because of {}.
I have places where {} is OK within architecture but I have to add additional check or add some attribute that I have to take care.

**Comment by iw2rmb** [comment]
@tjmonsi seems that setting zeroValue to `null` works as well as `dom-if`. Awesome!

**Comment by 43081j** [comment]
I would not recommend altering your copy of polymerfire locally.

This is because of #33.

TLDR: it currently resolves `null` values to `{}` to make bindings easier/more friendly.

**Comment by iw2rmb** [comment]
At least dom-if is not working with `{}`. In general I'd like to have `undefined` before binding and `null` if path is empty. And Yes, there are cases when default `{}` makes job easier. May be there should be parameter that handles behavior of zeroValue?

**Comment by 43081j** [comment]
Please discuss in #33 as this is a duplicate issue. You can see my thoughts and @mbleigh's in there also :)
